packages/@react-aria/actiongroup/src/ActionGroupKeyboardDelegate.ts

```diff
@@ -17,37 +17,93 @@ import {Key} from 'react';
 export class ActionGroupKeyboardDelegate<T> implements KeyboardDelegate {
   private collection: Collection<T>;
   private flipDirection: boolean;
+  private disabledKeys: Set<Key>;
 
-  constructor(collection: Collection<T>, direction: Direction, orientation: Orientation) {
+  constructor(collection: Collection<T>, direction: Direction, orientation: Orientation, disabledKeys: Set<Key> = new Set()) {
     this.collection = collection;
     this.flipDirection = direction === 'rtl' && orientation === 'horizontal';
+    this.disabledKeys = disabledKeys;
   }
 
   getKeyLeftOf(key: Key) {
-    return this.flipDirection 
-      ? this.collection.getKeyAfter(key) || this.getFirstKey() 
-      : this.collection.getKeyBefore(key) || this.getLastKey();
+    if (this.flipDirection) {
+      key = this.collection.getKeyAfter(key);
+      while (key) {
+        if (!this.disabledKeys.has(key)) {
+          return key;
+        }
+        key = this.collection.getKeyAfter(key);
+      }
+      return key || this.getFirstKey();
+    } else {
+      key = this.collection.getKeyBefore(key);
+      while (key) {
+        if (!this.disabledKeys.has(key)) {
+          return key;
+        }
+        key = this.collection.getKeyBefore(key);
+      }
+      return key || this.getLastKey();
+    }
```

Comment:
Could we make a couple helper functions for this and reuse them across all of these methods, e.g. `getNextKey`/`getPrevKey`?